src/main/java/com/github/sshaddicts/lucrecium/neuralNetwork/RichNeuralNet.java

```diff
@@ -1,96 +1,113 @@
     MultiLayerNetwork network;
 
-    private int layers = 0;
+    private int classNumber = 2;
+
+    public RichNeuralNet(int classNumber) {
+        this.classNumber = classNumber;
+    }
 
     public MultiLayerNetwork getNet() {
         return network;
     }
 
-    public void addLayer(Layer layer){
+    public void init(MultiLayerNetwork net) {
+        this.network = net;
     }
 
-    public void init(int inputSize, int hiddenNumber, int hiddenSize, int classNumber) {
-        layers = hiddenNumber + 1;
-        NeuralNetConfiguration.Builder builder = new NeuralNetConfiguration.Builder();
-
-        builder.iterations(ITERATIONS);
-        builder.learningRate(LEARNING_RATE);
-        builder.useDropConnect(USE_DROP_CONNECT);
-        builder.optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT);
+    public void init(int numRows, int numCols) {
+        System.out.println(String.format("using %d as input size", numRows * numCols));
+        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
+                .seed(123) //include a random seed for reproducibility
+                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT) // use stochastic gradient descent as an optimization algorithm
+                .iterations(1)
+                .activation(Activation.RELU)
+                .weightInit(WeightInit.XAVIER)
+                .learningRate(LEARNING_RATE) //specify the learning rate
+                .updater(new Nesterovs(0.98))
+                .regularization(true).l2(LEARNING_RATE * 0.005) // regularize learning model
+                .list()
+                .layer(0, new DenseLayer.Builder() //create the first input layer.
+                        .nIn(numRows * numCols)
+                        .nOut(500)
+                        .build())
+                .layer(1, new DenseLayer.Builder() //create the second input layer
+                        .nIn(500)
+                        .nOut(100)
+                        .build())
+                .layer(2, new OutputLayer.Builder(LossFunction.NEGATIVELOGLIKELIHOOD) //create hidden layer
+                        .activation(Activation.SOFTMAX)
+                        .nIn(100)
+                        .nOut(classNumber)
+                        .build())
+                .setInputType(InputType.convolutionalFlat(numRows, numCols, 1))
+                .pretrain(false).backprop(true) //use backpropagation to adjust weights
+                .build();
+
+        network = new MultiLayerNetwork(conf);
+        network.init();
+    }
 
-        builder.biasInit(0);
-        builder.miniBatch(MINI_BATCH);
+    public void train(DataSetIterator data) {
+        network.fit(data);
+    }
 
-        NeuralNetConfiguration.ListBuilder listBuilder = builder.list();
+    public void eval(INDArray input, INDArray actual) {
+        INDArray output = network.output(input);
+        eval = new Evaluation(classNumber);
```

Comment:
why are you redefining property of class? can `eval` be local immutable variable?